hack/debug-network.sh

```diff
@@ -0,0 +1,574 @@
+#!/bin/bash
+
+# echoes args to stderr and exits
+die () {
+    echo "$*" 1>&2
+    exit 1
+}
+
+# echoes the command provided as $@ and then runs it
+echo_and_eval () {
+    echo "> $*"
+    echo ""
+    eval "$@"
+}
+
+# runs the command provided as $@, and either returns silently with
+# status 0 or else logs an error message with the command's output
+try_eval () {
+    tmpfile=`mktemp`
+    if ! eval "$@" >& $tmpfile; then
```

Comment:
Lots of mismatched indents... use tabs for indentation, spaces for alignment past that